### PR TITLE
chore: remove zenkaku spaces

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

remove zenkaku spaces

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- Documentation: Removed zenkaku spaces (full-width spaces) from comments in various code files, improving code readability without affecting logic or functionality.
- Refactor: Modified comments in code files related to the behavior path goal planner module and behavior path planner, improving clarity and addressing comment suggestions regarding interface design. No impact on logic or functionality.
- Documentation: Updated parameter description in `goal_searcher.hpp` to exclude "freespace pull over" from the efficient order of the pull-over planner along lanes. No impact on logic or functionality.
- Style: Removed zenkaku spaces from comments in `BehaviorPathPlannerNode` class and `MissionPlanner` class. No impact on logic or functionality.

### CODE REVIEW

File: control/control_validator/src/utils.cpp [hunk 1]
```diff
@@ -119,7 +119,7 @@ Trajectory alignTrajectoryWithReferenceTrajectory(
 
   // If last point of predicted_trajectory is behind of end of trajectory, erase points which are
   // behind trajectory last point and insert pNew along the predicted_trajectory
-  // predicted_trajectory:   　　　　p1-----//------pN-2-----pN-1-----pN
+  // predicted_trajectory:           p1-----//------pN-2-----pN-1-----pN
   // trajectory:                     t1-----//-----tN-1--tN
   // ↓
   // predicted_trajectory:           p1-----//------pN-2-pNew
```
Comment (kosuke55) [inline control/control_validator/src/utils.cpp:125]: @kyoichi-sugahara 
is this alignment ok?

**Review by rej55 — APPROVED** [review]
Nice

File: control/control_validator/src/utils.cpp (hunk 1 above)
Comment (kyoichi-sugahara) [reply control/control_validator/src/utils.cpp:125]: perfect!!!
zenkaku lol

**Review by kyoichi-sugahara — APPROVED** [review]

**Review by soblin — APPROVED** [review]
Thank you so much this is really helpful !

**Review by miursh — APPROVED** [review]

**Review by ktro2828 — APPROVED** [review]
LGTM about the updates for `map_based_prediction`

**Review by yukkysaito — APPROVED** [review]

**Review by YoshiRi — COMMENTED** [review]
LGTM for tracking/prediction modules

**Review by satoshi-ota — APPROVED** [review]

**Review by scepter914 — APPROVED** [review]

**Review by takayuki5168 — APPROVED** [review]

**Review by TakaHoribe — APPROVED** [review]

**Review by shmpwk — APPROVED** [review]

**Pull request merged**